tdrs-backend/tdpservice/users/migrations/0012_add_initial_super_user.py

```diff
@@ -0,0 +1,47 @@
+import os
+import logging
+from django.contrib.auth.hashers import make_password
+from django.db import migrations
+from django.utils import timezone
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+
+class Migration(migrations.Migration):
+    dependencies = [
+        ('users', '0011_auto_20210108_1741'),
+    ]
+
+    def generate_superuser(apps, schema_editor):
+        # set the environment variable to the username of the
+        # initial superuser
+        su_username = os.environ.get('DJANGO_SU_NAME')
```

Comment:
In the event a local developer runs this migration before setting the environment variable this migration will create a User record with a blank string for a username:
```
In [1]: User.objects.get(username='')
Out[1]: <User: >
```
The user does get warned of this on startup:
```
❯ tdrs-backend-compose up --build -d                                                                                                                                                                            ─╯
WARNING: The DJANGO_SU_NAME variable is not set. Defaulting to a blank string.
Creating network "tdrs-backend_default" with the default driver
```
However, would it make sense to exit this migration with an error in the event that isn't set so we don't get an orphaned user record?

Reply:
Yes, it should throw an error